src/templates/base/2019/base.html

```diff
@@ -385,6 +385,43 @@
   {% endmacro %}
 {% endif %}
 
+{%- macro figure(
+  id=0,
+  image="",
+  caption="",
+  alt=caption,
+  description="",
+  sheets_url="",
+  width=600,
+  height=371,
+  data_width=width,
+  data_height=height,
+  content="",
+  classes=""
+  )
+%}
+<figure id="fig-{{ id }}">
+  {%- if image != "" %}
+  <a href="/static/images/{{ year }}/{{ metadata.get('chapter') }}/{{ image }}">
+    {%- if sheets_url != "" %}
+    <img src="/static/images/{{ year }}/{{ metadata.get('chapter') }}/{{ image }}" alt="{{ caption|striptags }}" aria-labelledby="fig{{ id }}-caption" aria-describedby="fig{{ id }}-description" width="{{ width }}" height="{{ height }}" data-width="{{ data_width }}" data-height="{{ data_height }}" data-seamless="" data-frameborder="0" data-scrolling="no" data-iframe="{{ sheets_url|safe }}" loading="lazy" />
+    {%- else %}
+    <img src="/static/images/{{ year }}/{{ metadata.get('chapter') }}/{{ image }}" alt="{{ caption|striptags }}" aria-labelledby="fig{{ id }}-caption" aria-describedby="fig{{ id }}-description" width="{{ width }}" height="{{ height }}" loading="lazy" />
+    {%- endif %}
+  </a>
+  <button hidden="" class="fig-description-button" aria-expanded="false" aria-controls="fig{{ id }}-description" data-show-text="{{ show_description(id) }}" data-hide-text="{{ hide_description(id) }}">{{ show_description(id) }}</button>
+  <div id="fig{{ id }}-description" class="visually-hidden">{{ description|safe }}</div>
+  {%- elif content != "" %}
+    {%- if classes != "" %}
+  <div class="{{ classes }}">{{ content }}</div>
+    {%- else %}
+  <div>{{ content }}</div>
+    {%- endif %}
```

Comment:
I am not sure if the linter will be happy with an empty class value like `class=""` (which is perfectly valid), if it is then we can avoid conditional in the template. Another possibility is to always have a default class value.

```suggestion
  <div class="{{ classes }}">{{ content }}</div>
```

Reply:
Yeah good idea. Think will always be a class for these so will just go with this simple one.